src/engraving/layout/layoutsystem.cpp

```diff
@@ -727,6 +727,15 @@ void LayoutSystem::layoutSystemElements(const LayoutOptions& options, LayoutCont
                 Beam* b = cr->beam();
                 b->layout();
             }
+            if (!cr->isChord()) {
+                continue;
+            }
+            for (Chord* grace : toChord(cr)->graceNotes()) {
+                Beam* beam = grace->beam();
+                if (beam && !beam->elements().empty() && beam->elements().front() == grace) {
+                    beam->layout();
+                }
+            }
```

Comment:
I suddenly remember that I've seen very similar code in `LayoutMeasure::getNextMeasure`. Would it make sense to make this one exactly the same as that? For example, shouldn't we use `LayoutBeams::isTopBeam` here?

Or maybe even better: add a static method in LayoutBeams that can be called both here and in `LayoutMeasure::getNextMeasure`.

Reply:
Oops, didn't entirely mean to approve already... anyway, I think this will be the last comment :)

Reply:
Done, I've put that into a dedicated method 👍